/*
 * This file is part of the VSS-SDK project.
 *
 * This Source Code Form is subject to the terms of the GNU GENERAL PUBLIC LICENSE,
 * v. 3.0. If a copy of the GPL was not distributed with this
 * file, You can obtain one at http://www.gnu.org/licenses/gpl-3.0/.
 */

//! Aqui, encontra-se a definição do pacote de comandos de um time no formato da biblioteca Google Protocol Buffers (protobuf).
//! Para mais informações sobre protobuf: https://developers.google.com/protocol-buffers/
//! Esse pacote é responsável por carregar as informações de comando de um time de futebol de robôs.
//! Essa pacote SOMENTE É UTILIZADO para enviar os comandos para robôs virtuais no VSS-Simulator
//! Dentre essas informações do pacote existem: ID do comando, se o time é amarelo ou azul, uma lista dinâmica de comandos
//! e um código de situação. Para utilizar esse protocolo, o mesmo deve ser précompílado para uma linguagem especifica
//! gerando um trecho de código que deve ser referenciado.

//! Aqui definisse um package, que na pratica gera um namespace ((C++) -> (!) using namespace vss_command;
package vss_command;


//! Essa mensagem é uma submensagem do pacote Global_Commands que de fato é enviado. Essa informação é utilizada para
//! represetar as velocidades das rodas direita e esquerda de um robô.
message Robot_Command{
	required float left_vel = 1;	//! Velocidade da roda esquerda
	required float right_vel = 2;	//! Velocidade da roda direita
}

//! Essa é a mensagem que de fato é enviada.
message Global_Commands{
	repeated Robot_Command	robot_commands = 2;		//! Lista dinâmica de comandos. Sendo sempre 3, devido a quantidade robôs
}
